pkg/controllers/vdb/status_reconciler.go

```diff
@@ -196,6 +196,14 @@ func (s *StatusReconciler) calculateSubclusterStatus(ctx context.Context, sc *va
 		if !ok {
 			continue
 		}
+		stsSize := sc.GetStsSize(s.Vdb)
+		if stsSize == 0 && stsSize != sc.Size {
+			s.setSubclusterStatusWhenShutdown(podIndex, curStat)
```

Comment:
If we want to skip the value assignment, why do we need to change the value to false on UpNode, installed, AddedToDB?

Reply:
In case they were true before statefulset's scale down. After scale down those fields will always be false because the pods have been deleted.